src/sentry/tasks/post_process.py

```diff
@@ -21,13 +21,21 @@
 
 if TYPE_CHECKING:
     from sentry.eventstore.models import Event
+    from sentry.eventstream.base import GroupState
 
 logger = logging.getLogger("sentry")
 
-
 locks = LockManager(build_instance_from_options(settings.SENTRY_POST_PROCESS_LOCKS_BACKEND_OPTIONS))
 
 
+class PostProcessJob(TypedDict, total=False):
+    event: "Event"
+    group_state: "GroupState"
+    is_reprocessed: bool
+    has_reappeared: bool
+    has_alert: bool
```

Comment:
Would it make sense to introduce a data class for every function that was extracted?

Now that we have separate functions for different operations, I can see us adding tests for each function to verify correctness without having to test the entire `post_process_group` function and validate side-effects.

Reply:
Does it make sense to make `PostProcessJob` a `@dataclass` instead of a `TypedDict`?

Reply:
nit: maybe name this `PostProcessGroupContext`?

Reply:
sure. don't see why not. don't know too much about the benefits of either approach though

Reply:
actually it might be easier to keep it a `TypedDict` for now until we get settled on the logic/design, if that's ok